src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { Todo } from '../../App';
+
+type Props = {
+  todos: Todo[];
+  updateTodoList: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+export const TodoForm: React.FC<Props> = ({ todos, updateTodoList }) => {
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
+
+  const addedTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
+  const preparedObject = {
+    id: addedTodoId,
+    title: title,
+    completed: false,
+    userId: currentUserId,
+  };
+
+  const isUserSelected = () => (currentUserId ? true : false);
+  const isTitleValid = () => (title ? true : false);
+
+  return (
+    <form
+      action="/api/todos"
+      method="POST"
+      onSubmit={event => {
+        event.preventDefault();
+
+        if (currentUserId && title) {
+          todos.push(preparedObject);
```

Comment:
Directly mutating the `todos` array with `todos.push(preparedObject)` is not recommended as it can lead to unexpected behavior. Instead, create a new array with the new todo added, e.g., `updateTodoList([...todos, preparedObject]);`.